pkg/sso/SSOLoginService.go

```diff
@@ -0,0 +1,322 @@
+		return nil, err
+	}
+	// Rollback tx on error.
+	defer tx.Rollback()
+
+	configDataByte, err := json.Marshal(request.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	existingModel, err := impl.ssoLoginRepository.GetActive()
+	if err != nil {
+		impl.logger.Errorw("error in creating new sso login config", "error", err)
+		return nil, err
+	}
+	existingModel.Active = false
+	existingModel.UpdatedOn = time.Now()
+	existingModel.UpdatedBy = request.UserId
+	_, err = impl.ssoLoginRepository.Update(existingModel, tx)
+	if err != nil {
+		impl.logger.Errorw("error in creating new sso login config", "error", err)
+		return nil, err
+	}
+
+	model := &repository.SSOLoginModel{
+		Name:   request.Name,
+		Config: string(configDataByte),
+	}
+	model.Active = true
+	model.CreatedBy = request.UserId
+	model.UpdatedBy = request.UserId
+	model.CreatedOn = time.Now()
+	model.UpdatedOn = time.Now()
+	_, err = impl.ssoLoginRepository.Create(model, tx)
+	if err != nil {
+		impl.logger.Errorw("error in creating new sso login config", "error", err)
+		return nil, err
+	}
+
+	//TODO- update argocd-cm
+	clusterBean, err := impl.clusterService.FindOne(cluster.ClusterName)
+	if err != nil {
+		return nil, err
+	}
+	cfg, err := impl.envService.GetClusterConfig(clusterBean)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := impl.K8sUtil.GetClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+	updateSuccess := false
+	retryCount := 0
+	for !updateSuccess && retryCount < 3 {
+		retryCount = retryCount + 1
+
+		cm, err := impl.K8sUtil.GetConfigMapFast(ArgocdConfigMapNamespace, ArgocdConfigMap, client)
+		if err != nil {
+			return nil, err
+		}
+		updatedData, err := impl.updateSSODexConfigOnAcdConfigMap(request.Config)
+		if err != nil {
+			return nil, err
+		}
+		data := cm.Data
+		data["dex.config"] = updatedData["dex.config"]
+		cm.Data = data
+		_, err = impl.K8sUtil.UpdateConfigMapFast(ArgocdConfigMapNamespace, cm, client)
+		if err != nil {
+			impl.logger.Warnw(" config map failed", "err", err)
+			continue
+		}
+		if err == nil {
+			impl.logger.Warnw(" config map apply succeeded", "on retryCount", retryCount)
+			updateSuccess = true
+		}
+	}
+	if !updateSuccess {
+		return nil, fmt.Errorf("resouce version not matched with config map attemped 3 times")
+	}
+
+	// TODO - END
+
+	err = tx.Commit()
+	if err != nil {
+		return nil, err
+	}
+	return request, nil
+}
+
+func (impl SSOLoginServiceImpl) UpdateSSOLogin(request *bean.SSOLoginDto) (*bean.SSOLoginDto, error) {
+	dbConnection := impl.userRepository.GetConnection()
+	tx, err := dbConnection.Begin()
+	if err != nil {
+		return nil, err
+	}
+	// Rollback tx on error.
+	defer tx.Rollback()
+
+	configDataByte, err := json.Marshal(request.Config)
+	if err != nil {
+		return nil, err
+	}
+	model, err := impl.ssoLoginRepository.GetById(request.Id)
+	if err != nil {
+		impl.logger.Errorw("error in update new sso login config", "error", err)
+		return nil, err
+	}
+	model.Url = request.Url
```

Comment:
code duplicacy from previous method

Reply:
> code duplicacy from previous method

create and update share some same piece code, yes separated common func requred for that.